### PR TITLE
feat(app): include play service utils

### PR DESCRIPTION
**Pull request by russellwheatley**
fixes this issue: https://invertase.canny.io/react-native-firebase/p/playservicesavailability. 

### Checklist

- [ ] Supports `Android`
- [ ] Supports `iOS`
- [ ] `e2e` tests added or updated in packages/**/e2e
- [ ] Flow types updated
- [ ] Typescript types updated

### Release Plan

[CATEGORY][type] [LOCATION] - Message

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

File: packages/app/lib/utils/index.js [hunk 1]
```diff
@@ -31,6 +31,41 @@ class FirebaseUtilsModule extends FirebaseModule {
     return this.native.isRunningInTestLab;
   }
 
+  get androidPlayServices() {
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]: This use to be called `playServicesAvailability` - the example above in the quick start also uses `playServicesAvailability`

File: packages/app/lib/utils/index.js [hunk 2]
```diff
@@ -31,6 +31,41 @@ class FirebaseUtilsModule extends FirebaseModule {
...
+    if (isIOS) {
+      return {};
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]:
```suggestion
      return {
        isAvailable: true,
        status: 0,
      };
```

Default value for iOS?

File: packages/app/lib/utils/index.js [hunk 3]
```diff
@@ -31,6 +31,41 @@ class FirebaseUtilsModule extends FirebaseModule {
...
+    }
+    return this.native.androidPlayServices;
+  }
+
+  getPlayServicesStatus() {
+    if (isIOS) {
+      return false;
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]: Should this not be; 
```suggestion
      return Promise.resolve(null);
```

File: packages/app/lib/utils/index.js [hunk 4]
```diff
@@ -31,6 +31,41 @@ class FirebaseUtilsModule extends FirebaseModule {
...
+    }
+    return this.native.androidGetPlayServicesStatus();
+  }
+
+  promptForPlayServices() {
+    if (isIOS) {
+      return false;
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]: Should this not be; 
```suggestion
      return Promise.resolve(null);
```

File: packages/app/lib/utils/index.js [hunk 5]
```diff
@@ -31,6 +31,41 @@ class FirebaseUtilsModule extends FirebaseModule {
...
+    }
+    return this.native.androidPromptForPlayServices();
+  }
+
+  makePlayServicesAvailable() {
+    if (isIOS) {
+      return false;
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]: Should this not be; 
```suggestion
      return Promise.resolve(null);
```

File: packages/app/lib/utils/index.js [hunk 6]
```diff
@@ -31,6 +31,41 @@ class FirebaseUtilsModule extends FirebaseModule {
...
+    }
+    return this.native.androidMakePlayServicesAvailable();
+  }
+
+  resolutionForPlayServices() {
+    if (isIOS) {
+      return false;
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]: Should this not be; 
```suggestion
      return Promise.resolve(null);
```

File: packages/app/lib/utils/index.js (hunk 2 above)
Comment (russellwheatley) [reply packages/app/lib/utils/index.js]: Good catch, couldn't find it in v6 docs so didn't want it to break hence returning `{}`.

File: packages/app/lib/utils/index.js (hunk 1 above)
Comment (russellwheatley) [reply packages/app/lib/utils/index.js]: Thanks, Mike. Sorting out types now 👍

**Comment by Ehesp** [comment]
One to document in the new docs - maybe under app?

**Comment by russellwheatley** [comment]
@Ehesp Agreed. I've already put the `utils` under `app` usage. Once this clears , I'll put in new docs 👍

File: packages/app/lib/utils/index.js [hunk 7]
```diff
@@ -31,6 +31,44 @@ class FirebaseUtilsModule extends FirebaseModule {
...
+  makePlayServicesAvailable() {
+    if (isIOS) {
+      return Promise.resolve(null);
+    }
+    return this.native.androidMakePlayServicesAvailable();
+  }
+
+  resolutionForPlayServices() {
+    if (isIOS) {
+      return Promise.resolve(null);
```
Comment (Salakar) [inline packages/app/lib/utils/index.js]: Your types now say these return undefined on iOS; updated below to match;

```suggestion
      return Promise.resolve();
    }
    return this.native.androidGetPlayServicesStatus();
  }

  promptForPlayServices() {
    if (isIOS) {
      return Promise.resolve();
    }
    return this.native.androidPromptForPlayServices();
  }

  makePlayServicesAvailable() {
    if (isIOS) {
      return Promise.resolve();
    }
    return this.native.androidMakePlayServicesAvailable();
  }

  resolutionForPlayServices() {
    if (isIOS) {
      return Promise.resolve();
```

File: packages/app/lib/index.d.ts [hunk 8]
```diff
@@ -379,6 +460,63 @@ export namespace Utils {
...
+
+    /**
+     * Returns true if this app is running inside a Firebase Test Lab environment. Always returns false on iOS.
+     *
+     * @android
+     */
+    playServicesAvailability: PlayServicesAvailability;
+    /**
+     * If true, allows the device to collect analytical data and send it to
+     * Firebase. Useful for GDPR.
```
Comment (Salakar) [inline packages/app/lib/index.d.ts]: Copy pasta derp 👀

File: packages/app/lib/index.d.ts [hunk 9]
```diff
@@ -379,6 +460,63 @@ export namespace Utils {
...
+     *
+     * #### Example
+     *
+     * ```js
+     * await firebase.utils().makePlayServicesAvailable();
+     * ```
+     *
+     * @android Android only - iOS returns null
+     */
+
```
Comment (Salakar) [inline packages/app/lib/index.d.ts]:
```suggestion
```

File: packages/app/lib/index.d.ts [hunk 10]
```diff
@@ -379,6 +460,63 @@ export namespace Utils {
...
+     * #### Example
+     *
+     * ```js
+     * await firebase.utils().promptForPlayServices();
+     * ```
+     *
+     * @android Android only - iOS returns null
+     */
+    promptForPlayServices(): Promise<void>;
+    /**
```
Comment (Salakar) [inline packages/app/lib/index.d.ts]:
```suggestion

    /**
```

File: packages/app/lib/index.d.ts [hunk 11]
```diff
@@ -379,6 +460,63 @@ export namespace Utils {
...
+     * #### Example
+     *
+     * ```js
+     * await firebase.utils().makePlayServicesAvailable();
+     * ```
+     *
+     * @android Android only - iOS returns null
+     */
+    makePlayServicesAvailable(): Promise<void>;
+    /**
```
Comment (Salakar) [inline packages/app/lib/index.d.ts]:
```suggestion

    /**
```

File: packages/app/lib/index.d.ts [hunk 12]
```diff
@@ -379,6 +460,64 @@ export namespace Utils {
...
+     * #### Example
+     *
+     * ```js
+     * const PlayServicesAvailability = await firebase.utils().getPlayServicesStatus();
+     * ```
+     *
+     * @android Android only - iOS returns null
+     */
+    getPlayServicesStatus(): Promise<PlayServicesAvailability>;
+    /**
```
Comment (Salakar) [inline packages/app/lib/index.d.ts]:
```suggestion

    /**
```

File: packages/app/lib/index.d.ts [hunk 13]
```diff
@@ -379,6 +465,66 @@ export namespace Utils {
      * @android
      */
     isRunningInTestLab: boolean;
+
+    /**
+     * Returns true if this app is running inside a Firebase Test Lab environment. Always returns false on iOS.
```
Comment (Salakar) [inline packages/app/lib/index.d.ts]: Copy pasta derp 👀

File: packages/app/lib/utils/index.js (hunk 7 above)
Comment (russellwheatley) [reply packages/app/lib/utils/index.js]: I'm confused, I thought my types for iOS were returning `null` 😅? If you prefer `undefined` I'll change accordingly 👍

Comment (Salakar) [reply packages/app/lib/utils/index.js]: Types only mention void (undefined), e.g. `resolutionForPlayServices(): Promise<void>;`:

image

If you change this then the `@android` docs tag on each probably also needs updating

Comment (russellwheatley) [reply packages/app/lib/utils/index.js]: Ah, I see. Cool, I'll update to  :`Promise<void | null>` for the types then 👍

**Pull request closed without merging**